nixos/modules/services/networking/bee/bee-clef.nix

```diff
@@ -0,0 +1,196 @@
+
+  makeSystemdTmpfilesEntry = id:
+    let
+      dataDir = deriveDataDir id;
+    in
+      [
+        "d '${dataDir}/'         0750 ${cfg.user} ${cfg.group}"
+        "d '${dataDir}/keystore' 0700 ${cfg.user} ${cfg.group}"
+      ];
+
+  makeSystemdServiceEntry = id:
+    let
+      dataDir        = deriveDataDir id;
+      namePostfix    = if id == null then "" else "-${toString id}";
+    in {
+      name = "bee-clef${namePostfix}";
+      value = {
+        path = [
+          # these are needed for the ensure-clef-account script
+          pkgs.coreutils
+          pkgs.gnused
+          pkgs.gawk
+        ];
+
+        wants    = [ "network.target" ];
+        wantedBy = [ "bee${namePostfix}.service" "multi-user.target" ];
+
+        serviceConfig = {
+          Nice = cfg.daemonNiceLevel;
+          NoNewPrivileges = true;
+          User = cfg.user;
+          Group = cfg.group;
+          ExecStartPre = ''${cfg.package}/share/bee-clef/ensure-clef-account'';
+          ExecStart = "${cfg.package}/share/bee-clef/bee-clef-service start";
+          ExecStop ="${cfg.package}/share/bee-clef/bee-clef-service stop";
+          Environment = [
+            #''"AUDIT_LOG_FILE=${cfg.auditLogDir}/clef-audit-log${namePostfix}.log"''
+            ''"CONFIG_DIR=${cfg.package}/share/bee-clef/"''
+            ''"DATA_DIR=${dataDir}"''
+            ''"PASSWORD_FILE=${cfg.passwordFile}"''
+          ];
+          Restart = "on-failure";
+          RestartSec = 5;
+        } // optionalAttrs (cfg.daemonIOSchedulingClass != null) {
+          IOSchedulingClass = cfg.daemonIOSchedulingClass;
+        } // optionalAttrs (cfg.daemonIOSchedulingPriority != null) {
+          IOSchedulingPriority = cfg.daemonIOSchedulingPriority;
+        };
+      };
+    };
+
+in {
+  meta = {
+    maintainers = with maintainers; [ attila-lendvai ];
+  };
+
+  ### interface
+
+  options = {
+    services.bee-clef = {
+      package = mkOption {
+        type = types.package;
+        default = pkgs.bee-clef;
+        defaultText = "pkgs.bee-clef";
+        example = "pkgs.bee-clef-unstable";
+        description = "The package providing the bee-clef files for the service.";
+      };
+
+      dataDir = mkOption {
+        type = types.nullOr types.path;
+        default = "/var/lib/bee-clef";
+        description = ''
+          Base directory for bee-clef instances. Each instance will be under a subdirectory named `[0..n]`.
+        '';
+      };
+
+      passwordFile = mkOption {
+        type = types.nullOr types.path;
+        default = "/var/lib/bee-clef/password";
+        description = ''
+          Password file shared by each bee-clef instance. Will be generated from `/dev/urandom` when missing.
+        '';
+      };
+
+      instanceCount = mkOption {
+        type = types.ints.unsigned;
+        default = if config.services.bee.settings.clef-signer-enable or false
+                  then config.services.bee.instanceCount
+                  else 0;
```

Comment:
@SuperSandro2000 is this above valid, or "legal"? it works, and does what i want, but it breaks the compilation of the manual. i'm out of ideas how else to do this. i could merge the two services into one file/namespace, but that would hurt readability... i'd need to isolate a lot of names, like `dataDir`.

```
$ nix-build nixos/release.nix --show-trace -A manual.x86_64-linux
error: while evaluating anonymous function at /media/store/work/nixos/nixpkgs/nixos/release.nix:92:49, called from /media/store/work/nixos/nixpkgs/lib/attrsets.nix:310:43:
while evaluating 'hydraJob' at /media/store/work/nixos/nixpkgs/lib/customisation.nix:171:14, called from /media/store/work/nixos/nixpkgs/nixos/release.nix:92:57:
while evaluating the attribute 'drvPath' at /media/store/work/nixos/nixpkgs/lib/customisation.nix:188:13:
while evaluating the attribute 'drvPath' at /media/store/work/nixos/nixpkgs/lib/customisation.nix:155:13:
while evaluating the attribute 'buildCommand' of the derivation 'nixos-manual-html' at /media/store/work/nixos/nixpkgs/pkgs/build-support/trivial-builders.nix:7:7:
while evaluating the attribute 'buildCommand' of the derivation 'manual-olinkdb' at /media/store/work/nixos/nixpkgs/pkgs/build-support/trivial-builders.nix:7:7:
while evaluating the attribute 'buildCommand' of the derivation 'nixos-manual-combined' at /media/store/work/nixos/nixpkgs/pkgs/build-support/trivial-builders.nix:7:7:
while evaluating the attribute 'buildCommand' of the derivation 'generated-docbook' at /media/store/work/nixos/nixpkgs/pkgs/build-support/trivial-builders.nix:7:7:
while evaluating the attribute 'buildCommand' of the derivation 'options-docbook.xml' at /media/store/work/nixos/nixpkgs/pkgs/build-support/trivial-builders.nix:7:7:
while evaluating 'substFunction' at /media/store/work/nixos/nixpkgs/nixos/lib/make-options-doc/default.nix:28:19, called from /media/store/work/nixos/nixpkgs/nixos/lib/make-options-doc/default.nix:37:54:
while evaluating the attribute 'default' at /media/store/work/nixos/nixpkgs/lib/options.nix:187:44:
while evaluating 'scrubOptionValue' at /media/store/work/nixos/nixpkgs/lib/options.nix:206:22, called from /media/store/work/nixos/nixpkgs/lib/options.nix:187:54:
while evaluating 'isDerivation' at /media/store/work/nixos/nixpkgs/lib/attrsets.nix:323:18, called from /media/store/work/nixos/nixpkgs/lib/options.nix:207:8:
while evaluating the attribute 'default' at /media/store/work/nixos/nixpkgs/nixos/modules/services/networking/bee/bee-clef.nix:100:9:
while evaluating the attribute 'services.bee.settings.clef-signer-enable' at undefined position:
while evaluating 'g' at /media/store/work/nixos/nixpkgs/lib/attrsets.nix:294:19, called from undefined position:
while evaluating anonymous function at /media/store/work/nixos/nixpkgs/lib/modules.nix:139:72, called from /media/store/work/nixos/nixpkgs/lib/attrsets.nix:297:20:
while evaluating the attribute 'value' at /media/store/work/nixos/nixpkgs/lib/modules.nix:512:9:
while evaluating the option `services.bee.settings':
while evaluating the attribute 'mergedValue' at /media/store/work/nixos/nixpkgs/lib/modules.nix:544:5:
The option `services.bee.settings' is used but not defined.
```